src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore
+{
+	using System;
+	using System.Linq;
+
+	using Castle.MicroKernel.Lifestyle;
+	using Castle.MicroKernel.Registration;
+	using Castle.Windsor;
+
+	using Microsoft.AspNetCore.Builder;
+	using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Mvc;
+	using Microsoft.AspNetCore.Razor.TagHelpers;
+	using Microsoft.Extensions.DependencyInjection;
+
+	public static class CastleWindsorRegistrationExtensions
+	{
+		private static readonly string[] InvalidFrameworkConfigurationNamespaces =
+		{
+			"Microsoft.AspNetCore"
+			// Todo: Might need to add some more framework namespaces.
+		};
+
+		public static void AddCastleWindsor(this IServiceCollection services, IWindsorContainer container)
+		{
+			services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+			services.AddRequestScopingMiddleware(container.BeginScope);
+			services.AddCustomControllerActivation(container.Resolve);
+			services.AddCustomTagHelperActivation(container.Resolve);
+			services.AddCustomViewComponentActivation(container.Resolve);
+			container.Kernel.Resolver.AddSubResolver(new FrameworkConfigurationDependencyResolver(services));
+		}
+
+		public static void UseCastleWindsor<TStartup>(this IApplicationBuilder app, IWindsorContainer container)
+		{
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TStartup).Assembly).BasedOn<Controller>().LifestyleScoped());
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TStartup).Assembly).BasedOn<ViewComponent>().LifestyleTransient());
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TStartup).Assembly).BasedOn<TagHelper>().LifestyleTransient());
+		}
+
+		public static void UseCastleWindsorMiddleware<T>(this IApplicationBuilder app, IWindsorContainer container) where T : class, ICastleWindsorMiddleware
+		{
+			container.Register(Component.For<T>());
+			app.Use(async (context, next) => { await container.Resolve<T>().Invoke(context, next); });
+		}
+
+		public static void ValidateConfiguration(this IWindsorContainer container)
```

Comment:
This extension method sounds sort of misleading extending `IWindsorContainer` as it gets called looking like `container.ValidateConfiguration();`. Also do users have to call this manually as your example in the docs doesn't show it and only a unit test calls it.

Reply:
This is a helpful exception, that could hopefully be added to a test fixture somewhere. Did not want this getting in the way of having to register framework types(even though we say you should not) but in a test scenario it should light up red. The docs needs a bit updating here for sure. Let me write this up a bit better.

Reply:
Thanks for the description, I can see how this would be helpful for a unit test, I just expected that it would also get throw during runtime as most users especially those that are new aren't likely to be unit testing this stuff. But I could see there might be a reason someone really does need to register one of these types, and it is probably a good idea not to block it hard.